src/deepqmc/fit.py

```diff
@@ -182,31 +187,34 @@ def fit_wf(  # noqa: C901
             torch.isnan(p.grad).any() for p in wf.parameters() if p.grad is not None
         ):
             raise NanError(rs_batch)
-        loss = loss.sum()
+        loss = batch_sum(loss)
         if max_grad_norm is not None:
             clip_grad_norm_(wf.parameters(), max_grad_norm)
-        E_loc_mean, E_loc_var = weighted_mean_var(Es_loc, log_ws)
-        E_loc_err = torch.sqrt(E_loc_var / len(Es_loc))
+        E_loc_mean, E_loc_var = batch_weighted_mean_var(Es_loc, log_ws)
+        E_loc_err = torch.sqrt(E_loc_var / batch_len(Es_loc))
         lr = opt.state_dict()['param_groups'][0]['lr']
         if writer:
             writer.add_scalar('E_loc/mean', E_loc_mean, step)
             writer.add_scalar('E_loc/var', E_loc_var, step)
-            writer.add_scalar('E_loc/min', Es_loc.min(), step)
-            writer.add_scalar('E_loc/max', Es_loc.max(), step)
+            writer.add_scalar('E_loc/min', batch_min(Es_loc), step)
+            writer.add_scalar('E_loc/max', batch_max(Es_loc), step)
             writer.add_scalar('E_loc/err', E_loc_err, step)
```

Comment:
Yes, all processes are doing the logging. This does not take longer than logging only from a single process, because all other processes have to wait anyways, and reduces the necessairy code changes.